builtin/providers/github/data_source_github_user.go

```diff
@@ -0,0 +1,118 @@
+package github
+
+import (
+	"context"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceGithubUser() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGithubUserRead,
+
+		Schema: map[string]*schema.Schema{
+			"username": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"login": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"avatar_url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"gravatar_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"site_admin": &schema.Schema{
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"company": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"blog": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"email": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"bio": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"public_repos": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"public_gists": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"followers": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"following": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"created_at": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceGithubUserRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Refreshing Gitub User: %s", d.Id())
```

Comment:
On an initial run the `id` is `nil`, could change the log statement to print the `username` as it's a required parameter